lib/mdl/rules.rb

```diff
@@ -796,3 +796,85 @@ def check_blockquote(errors, elements)
     error_lines
   end
 end
+
+rule 'MD055', 'Tables should be well-formed' do
+  tags :tables
+  aliases 'table-format-validation'
+  check do |doc|
+    error_lines = []
+    tables = doc.find_type_elements(:table)
+    lines = doc.lines
+    num_lines = lines.length
+
+    tables.each do |table|
+      loc = table.options[:location]-1
+      header = lines[loc]
+
+      if header[0] != '|' || header[-1] != '|' || header.length == 1
+        # Tables should start and end with '|'
+        error_lines << (loc+1)
+      end
+
+      headings = header.strip().split('|').select{ |x| x.length > 0 }
```

Comment:
why only headings that are > 0 ? If I do `|foo||bar|` - there's 3 headings, one is just blank, no?

I guess because of a heading is blank, then you're implying that the whole column should be blank?

I'd sorta be OK with that, but that means that I could have 5 columns with one blank one in the heading and 17 columns on some row, as long as 13 of them were blank. That doesn't feel right to me. I'd rather just compare the total number of columns anywhere, regardless of if they're blank or not.

Reply:
I'll come back to this, because I'm not able to reproduce what I was having when I added this part 🤣 Part of the reason is I'm not able to understand what split() is exactly doing. 

```
'||'.split('|') => []
'|| '.split('|') => ["", "", " "]
' ||'.split('|') => [" "]
'| |'.split('|') => ["", " "]
```
Maybe I shouldn't use `split()`.

Reply:
OK yeah I can see why this is confusing. But that's only because markdown tables are a weird form of separation/join.

Think about any language, what happens when you join `['one', 'two', 'three']` on pipes? You get `one|two|three`. So any split function will assume that whatever is before the first separator is an item in the list, even if it's `nil`/`empty`. Additionally, it makes sense to take anything after the final separator as an item only if it exists since trailing commas/separators/etc. are common. So that takes care of most of the weirdness here.

Then there's a weird corner case with the "empty" case (`||`) which ... I can't explain.

With that in mind something like this should work:

```ruby
columns = line.strip.strip
# unlike most joined strings, we only want values AFTER the first
# separator
trash = columns.shift
unless trash.nil? or trash.empty?
  raise "wtf"
end
```

Reply:
To be honest, from the `split` and `join` in a language, I'd expect
```
join(split(str, sep), sep) == str
```
That would make
```
'||'.split('|') => ["", "", ""]
'|| '.split('|') => ["", "", " "]
' ||'.split('|') => [" ", "", ""]
'| |'.split('|') => ["", " ", ""]
```
But, I guess I'll try to stay away from this function for the time being.

Reply:
`split` is the right call here, for sure. And there's definitely a weird corner case in the `||` case which is silly to me. But anyway I think the above code will get you what you want